### PR TITLE
shipwright: 7.0.2 -> 7.1.1

### PR DESCRIPTION
**Pull request by MisterPeModder**
## Description of changes
Updating Shipwright from version 7.0.2 to 7.1.1
Removed the `lus-install-paths` and `soh-misc-otr-patches` patches, functionality is now present in upstream.

Changelog: https://www.shipofharkinian.com/changelog#sulu-bravo-7-1-1

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/shipwright/default.nix [hunk 1]
```diff
@@ -95,13 +95,13 @@ assert (lib.assertMsg (oot.enable || ootMq.enable) "At least one of 'oot.enable'
...
-  version = "7.0.2";
+  version = "7.1.1";
 
   src = fetchFromGitHub {
     owner = "harbourmasters";
     repo = "shipwright";
-    rev = version;
-    hash = "sha256-2VCcczGWSvp6hk8FTA1/T1E1KkrrvWyOdkEw8eiYYnY=";
+    rev = "7.1.1";
+    hash = "sha256-zgxJj65wKsQWvVxeCspyHG9YqoYqZxd6GrYptOA8Byk=";
```
Comment (cafkafk) [inline pkgs/games/shipwright/default.nix:104]: This should be `version`, no reason to hardcode it

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 258470` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>shipwright</li>
  </ul>
</details>

**Review by cafkafk — CHANGES_REQUESTED** [review]

**Review by cafkafk — COMMENTED** [review]
when running `nixpkgs-rewview pr 258470`

```
error: builder for '/nix/store/bcqhn5kgngj6mm0ryrq936h515p2hdxd-oot-debug.z64.drv' failed with exit code 1;
       last 10 log lines:
       > The supported variants are:
       >  - debug: Ocarina of Time Debug PAL GC (not Master Quest)
       >  - pal-gc: Ocarina of Time PAL GameCube (may lead to crashes and instability)
       >
       > This is optional if you have imported an Ocarina of Time Master Quest ROM.
       > If so, please set oot.enable to false and ootMq.enable to true.
       > If both are enabled, Ship of Harkinian will be built with both ROMs.
       >
       > ***
       >
       For full logs, run 'nix log /nix/store/bcqhn5kgngj6mm0ryrq936h515p2hdxd-oot-debug.z64.drv'.
error: 1 dependencies of derivation '/nix/store/09ra8lmbqhza0px9sfjvbbb90yqj5j1r-shipwright-7.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5s5ksxaighscndnmchnd2p34zc1x0df9-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/258470

1 package failed to build:
shipwright
```

This does seem to be on purpose thou

**Comment by Krutonium** [comment]
Hi, I would love to make a request!
So Shipwright supports mods, yes? I would love the ability to specify within Nix paths to .otr files to include in the mods directory of the derivation, for example https://evilgames.eu/texture-packs/oot-reloaded.htm - Which provides an HD texture pack for the game.  (In this case, https://evilgames.eu/texture-packs/files/oot-reloaded-v10.4.0-soh-otr_8x.7z as an example of what I'd like to include, but also the mod that adds DPad bindings for the masks)

**Comment by Krutonium** [comment]
https://github.com/NixOS/nixpkgs/pull/258470#pullrequestreview-1652074630

I get the same error, despite having an appropriate ROM in my nix store (PAL-GC)

File: pkgs/games/shipwright/default.nix (hunk 1 above)
Comment (MisterPeModder) [reply pkgs/games/shipwright/default.nix]: My bad. I changed the version for testing and forgot to revert this line.
It's fixed now

**Comment by MisterPeModder** [comment]
> when running `nixpkgs-rewview pr 258470`
> 
> ```
> error: builder for '/nix/store/bcqhn5kgngj6mm0ryrq936h515p2hdxd-oot-debug.z64.drv' failed with exit code 1;
>        last 10 log lines:
>        > The supported variants are:
>        >  - debug: Ocarina of Time Debug PAL GC (not Master Quest)
>        >  - pal-gc: Ocarina of Time PAL GameCube (may lead to crashes and instability)
>        >
>        > This is optional if you have imported an Ocarina of Time Master Quest ROM.
>        > If so, please set oot.enable to false and ootMq.enable to true.
>        > If both are enabled, Ship of Harkinian will be built with both ROMs.
>        >
>        > ***
>        >
>        For full logs, run 'nix log /nix/store/bcqhn5kgngj6mm0ryrq936h515p2hdxd-oot-debug.z64.drv'.
> error: 1 dependencies of derivation '/nix/store/09ra8lmbqhza0px9sfjvbbb90yqj5j1r-shipwright-7.1.1.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/5s5ksxaighscndnmchnd2p34zc1x0df9-review-shell.drv' failed to build
> 
> Link to currently reviewing PR:
> https://github.com/NixOS/nixpkgs/pull/258470
> 
> 1 package failed to build:
> shipwright
> ```
> 
> This does seem to be on purpose thou

The way this package is currently setup requires manual setup unfortunately.
Running `nix-store --add-fixed sha1 oot-debug.z64` did the trick for me, the file had to be called exactly `oot-debug.z64` tough. (sha1 cee6bc3c2a634b41728f2af8da54d9bf8cc14099).

**Comment by katexochen** [comment]
Closes https://github.com/NixOS/nixpkgs/pull/242259.

**Review by r-burns — APPROVED** [review]
Can confirm this runs with the proper fixed-path setup. Looks good, thanks!

Congrats on your first contribution to nixpkgs, and sorry for the slow turnaround on this :)

**Pull request merged**